### PR TITLE
Feature/associations

### PR DESCRIPTION
**Pull request by AmineDerbal**
In this pull request ♻️ 

- Create class Classroom with @label variable and update the association with the Student class 👍 
- Create a class Book with all variables and the setters and getters 👍 
- Create a class Rental with all variables and the setters and getters 👍 
- update the association between student and book using rental as an intermediate class 👍

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require_relative 'rentable'
+
+class Book
+  attr_accessor :title, :author
+
```
Comment (gemmen29) [inline book.rb:5]: - [ ] Kindly define the `@rentals` list as `attr_accessor` as you can access it inside the initialize method Rental class.

File: person.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 require_relative 'nameable'
 require_relative 'capitalize_decorator'
 require_relative 'trimmer_decorator'
+require_relative 'rentable'
 
 class Person < Nameable
   attr_accessor :name, :age
```
Comment (gemmen29) [inline person.rb]: - [ ] Kindly define the `@rentals` list as `attr_accessor` as you can access it inside the initialize method Rental class.

File: person.rb [hunk 3]
```diff
@@ -22,6 +24,10 @@ def correct_name
     @name
   end
 
+  def add_rental(date, person, book)
+    Rental.new(date, person, book)
+  end
```
Comment (gemmen29) [inline person.rb]: - [ ] [OPTIONAL] Kindly update this method to be like this:
```ruby
  def add_rental(book, date)
    Rental.new(date, self, book)
  end
```
You don't need to pass `person` as a parameter as you already have access to it on `self`.

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(date, person, book)
+    Rental.new(date, person, book)
+  end
```
Comment (gemmen29) [inline book.rb]: - [ ] [OPTIONAL] Kindly update this method to be like this:
```ruby
  def add_rental(person, date)
    Rental.new(date, person, self)
  end
```
You don't need to pass `book` as a parameter as you already have access to it on `self`.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @AmineDerbal, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (AmineDerbal) [reply book.rb:5]: done

File: person.rb (hunk 2 above)
Comment (AmineDerbal) [reply person.rb]: done

File: person.rb (hunk 3 above)
Comment (AmineDerbal) [reply person.rb]: done

File: book.rb (hunk 4 above)
Comment (AmineDerbal) [reply book.rb]: done

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @AmineDerbal,

✔️ Implementation of getters and setters

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**